Geometry/HcalTowerAlgo/src/HcalGeometry.cc

```diff
@@ -83,6 +83,27 @@ HcalGeometry::getValidDetIds( DetId::Detector det,
 		 ( HcalForward == subdet ? *m_hfIds.load() : *m_emptyIds.load() ) ) ) ) ) ;
 }
 
+const CaloCellGeometry* HcalGeometry::getGeometry(const DetId& id) const {
+#ifdef EDM_ML_DEBUG
+  std::cout << "HcalGeometry::getGeometry for " << HcalDetId(id) << "  " 
+	    << m_mergePosition << " ";
+#endif
+  if (!m_mergePosition) {
+#ifdef EDM_ML_DEBUG
+    std::cout << m_topology.detId2denseId(id) << " " 
+	      << cellGeomPtr(m_topology.detId2denseId(id)) << "\n";
+#endif
+    return cellGeomPtr(m_topology.detId2denseId(id)); 
```

Comment:
how about return getGeometryBase(m_topology.detId2denseId(id)) here and below.

Reply:
Done